packages/big-design/src/managers/alerts/manager.ts

```diff
@@ -0,0 +1,92 @@
+import { AlertProps } from '../../components/Alert';
+
+import { Subscriber, TypeMap } from './types';
+
+interface PrivateAlert extends AlertProps {
+  onClose(): void;
+}
+
+class AlertsManager {
+  private alerts: PrivateAlert[] = [];
+  private counter = 0;
+  private subscribers: Subscriber[] = [];
+  private readonly typeMap: TypeMap = {
+    error: 0,
+    warning: 1,
+    success: 2,
+    info: 3,
+  };
+
+  add(alert: AlertProps, dismissCallback?: () => void): string {
+    if (alert.key && this.containsKey(alert.key)) {
+      this.remove(alert.key);
+    }
+
+    const key = alert.key !== undefined ? alert.key : this.getUniqueId();
+    const onClose = () => {
+      if (typeof dismissCallback === 'function') {
+        dismissCallback(); // Should we return something with this?
+      }
+
+      if (typeof alert.onClose === 'function') {
+        alert.onClose();
+      }
+
+      this.remove(key);
+    };
```

Comment:
What was the reasoning behind keeping both `dismissCallback` and `onClose`?

Reply:
I was a little concerned about this too. I'm not sure we even need a `dismissCallback`, but we definitely want the `onClose`.

Reply:
We are going to remove this since it doesn't really make sense.